### DISCUSSION TITLE
Render Markdown, Mermaid, etc

### DISCUSSION THREAD

**Discussion by snoblenet** [chase/awrit#22] [Q&A] [answered]
Great tool! Could this be used to render Markdown, Mermaid, etc too?

**Comment by chase** [comment, marked as answer]
> Great tool! Could this be used to render Markdown, Mermaid, etc too?

There are a plenty of tools out there that should provide a live preview of Markdown/Mermaid in a browser, so I don't see why not.

You could use something like https://github.com/iamcco/markdown-preview.nvim more specifically, but it would require some extra setup to open a window in Kitty and then use `awrit` to open the preview URL

**Comment by byakuya-H** [comment]
I have made some attempts to configure Kitty and Neovim for making awrit as
markdown-preview.nvim preview browser.

Roughly speaking, I utilize kitten to remotely control kitty to launch awrit
in the terminal.

Initially, launch kitty listening to a specified socket using command `kitty
--listen-on /tmp/kitty.socket`; then configure markdown-preview.nvim as follows

```vimscript
function OpenMarkdownPreview (url)
	if has_key(environ(), "KITTY_LISTEN_ON") && system("command -v awrit") != ""
		"... some configuration for kitty to render better display, like `kitten ... load-config --override inactive_text_alpha=1.0`
		execute "silent ! kitten @ --to " . environ()["KITTY_LISTEN_ON"] . " launch awrit " . a:url
	else
		execute "silent ! firefox --new-window " . a:url
	endif
endfunction
let g:mkdp_browserfunc = 'OpenMarkdownPreview'
```

The configuration above will open a window and launch awrit. Alternatively use
command `"kitten ... launch --locatin=hsplit awrit " . a:url` to open a
horizontal split window.

Maybe some further configuration is needed.
